open_rarity/rarity_ranker.py

```diff
@@ -20,35 +27,75 @@ def rank_tokens(*, token_id_to_scores: dict[str, float]) -> dict[str, int]:
 
         Parameters
         ----------
-        token_ids_to_score : dict[str, float]
-            A dictionary of token ids (this can be whatever id you want, as long
-            as its unique) to the rarity score.
+        collection : Collection
+            Collection object with populated tokens
+        scorer: Scorer
+            Scorer instance
 
         Returns
         -------
-        dict[str, int]
-            A dictionary of token ids to their ranks. More specificially, it is the
-            provided tokens ids to the rank it has within the set of possible
-            tokens defined by @token_id_to_scores.
+        list[TokenRarity]
+            list of TokenRarity objects with score, rank and token information
+        """
+
+        if (
+            collection is None
+            or collection.tokens is None
+            or len(collection.tokens) == 0
+        ):
+            return collection
+
+        tokens = collection.tokens
+        scores: list[float] = scorer.score_tokens(collection, tokens=tokens)
+
+        # fail ranking if dimension of scores doesn't match dimension of tokens
+        assert len(tokens) == len(scores)
+
+        token_rarity_list: list[TokenRarity] = []
+        # augment collection tokes with score information
+        for idx, token in enumerate(tokens):
+            token_rarity_list.append(
+                TokenRarity(token=token, score=scores[idx])
+            )
+
+        return RarityRanker.rank_tokens(token_rarity_list)
+
+    @staticmethod
+    def rank_tokens(tokens: list[TokenRarity]) -> list[TokenRarity]:
```

Comment:
This shouldn't be `tokens` but `token_rarities` - otherwise it's confusing that it's not a `list[Token]`

Reply:
Can we change method name just to make it clear this modifies the input, otherwise we should make a copy of input which is inefficient? E.g.

`def set_ranks(token_rarities: list[TokenRarity]) -> list[TokenRarity]`

side note: If user's are using this one-off instead of rank_collection, they would need to construct list[TokenRarity] themselves. It makes me think of whether the scorer should also just return `list[TokenRarity] instead of list[score]`